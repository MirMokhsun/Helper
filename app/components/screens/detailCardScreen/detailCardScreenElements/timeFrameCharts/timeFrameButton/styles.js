import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        zIndex: 2,
    },
    text: {
        fontFamily: 'Roboto-Regular',
        fontWeight: 'normal',
        lineHeight: 24,
        fontSize: 12,
        color: '#9FA0B2',
    },
    activeText: {
        fontFamily: 'Roboto-Regular',
        fontWeight: 'bold',
        lineHeight: 24,
        fontSize: 12,
        color: '#000756',
    },
});