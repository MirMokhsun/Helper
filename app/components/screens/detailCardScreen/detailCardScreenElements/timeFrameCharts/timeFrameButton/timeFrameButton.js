import React, { PureComponent } from 'react';
import { View, Text, TouchableWithoutFeedback } from 'react-native';
import { styles } from './styles';

export class TimeFrameButton extends PureComponent {
    constructor(props) {
        super(props);

    }

    changeFrame = () => {
        const { setFrameFunction, frame } = this.props;
        setFrameFunction(frame);
    };

    render() {
        const { text, isActive, id } = this.props;
        return (
            <TouchableWithoutFeedback onPress={this.changeFrame} testID={`TimeFrameButton${id}ID`} accessibilityLabel={`TimeFrameButton${id}ID`}>
                <View style={styles.container} >
                    <Text style={[styles.text, isActive ? styles.activeText : null]}>{text}</Text>
                </View>
            </TouchableWithoutFeedback>
        );
    }
}

export default TimeFrameButton;