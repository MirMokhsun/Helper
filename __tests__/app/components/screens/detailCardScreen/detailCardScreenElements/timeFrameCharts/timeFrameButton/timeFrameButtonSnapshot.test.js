import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import { TimeFrameButton } from '../../../../../../../../app/components/screens/detailCardScreen/detailCardScreenElements/timeFrameCharts/timeFrameButton/timeFrameButton';

describe('TimeFrameButton', () => {
    test('Create snapshot TimeFrameButton with props isActive={false}', () => {
        const renderer = new ShallowRenderer()
        let timeFrameButton = renderer.render(<TimeFrameButton text={'1min'} isActive={false} setFrameFunction={() => { }} frame={'Minute'} />);
        expect(timeFrameButton).toMatchSnapshot();
    });
    test('Create snapshot TimeFrameButton with props isActive={true}', () => {
        const renderer = new ShallowRenderer()
        let timeFrameButton = renderer.render(<TimeFrameButton text={'1min'} isActive={true} setFrameFunction={() => { }} frame={'Minute'} />);
        expect(timeFrameButton).toMatchSnapshot();
    });
    test('test function changeFrame', () => {
        let frame = 'Minute';
        const setFrameFunction = () => {
            frame = 'Hour';
        };
        let timeFrameButton = renderer.create(<TimeFrameButton text={'1min'} isActive={false} setFrameFunction={setFrameFunction} frame={frame} />);
        timeFrameButton.getInstance().changeFrame();
        expect(frame).toEqual('Hour');
    });
});