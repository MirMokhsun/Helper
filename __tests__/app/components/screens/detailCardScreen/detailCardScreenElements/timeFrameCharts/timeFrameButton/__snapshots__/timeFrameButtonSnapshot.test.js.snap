// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeFrameButton Create snapshot TimeFrameButton with props isActive={false} 1`] = `
<TouchableWithoutFeedback
  accessibilityLabel="TimeFrameButtonundefinedID"
  onPress={[Function]}
  testID="TimeFrameButtonundefinedID"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "zIndex": 2,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#9FA0B2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 12,
            "fontWeight": "normal",
            "lineHeight": 24,
          },
          null,
        ]
      }
    >
      1min
    </Text>
  </View>
</TouchableWithoutFeedback>
`;

exports[`TimeFrameButton Create snapshot TimeFrameButton with props isActive={true} 1`] = `
<TouchableWithoutFeedback
  accessibilityLabel="TimeFrameButtonundefinedID"
  onPress={[Function]}
  testID="TimeFrameButtonundefinedID"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "zIndex": 2,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#9FA0B2",
            "fontFamily": "Roboto-Regular",
            "fontSize": 12,
            "fontWeight": "normal",
            "lineHeight": 24,
          },
          Object {
            "color": "#000756",
            "fontFamily": "Roboto-Regular",
            "fontSize": 12,
            "fontWeight": "bold",
            "lineHeight": 24,
          },
        ]
      }
    >
      1min
    </Text>
  </View>
</TouchableWithoutFeedback>
`;
